### ISSUE TITLE
Problema URL (https to http)

### ISSUE THREAD

**Issue by C0nrad92** [m3m3nto/giaa#10]
Ciao, scusa se ti disturbo di nuovo, ma sto riscontrando un altro problema.

Questa volta dovrei fare il ping di update e dopo aver seguito tutte le guide ed essere diventato proprietario ho inserito il link, come ho sempre già fatto senza problemi.

Adesso però, dopo aver inserito il link in "https://", lui automaticamente nella lista sotto va a cambiare l'url e lo riscrive come "http://". Così da un errore perché non ho verificato la proprietà del protocollo http, ma solo di quello in https (Dato che su GSC ho solo il sito in https)!

Perché va autonomamente a modificare l'url che io ho inserito?

Grazie ancora per la disponibilità!

**Comment by m3m3nto** [comment]
Possibile che nella configurazione tu abbia inserito http invece che https? Se è così nell'area config prova a modificare la URL coerentemente con la property verificata su GSC.

**Comment by C0nrad92** [comment]
Ciao, grazie per la risposta!

No, il problema non è quello purtroppo. 

Sto invece notando una cosa... 

Su GSC la proprietà è inserita con il trailing slash, lo slash finale, quindi ".it/", ma se inserisco nella configurazione di GIAA il dominio con ".it/"mi dice "GSC url property not found", mentre se lo configuro con ".it" resta in pending e nella console mi da il solito errore.

Potrebbe essere un'altra limitazione del tool o sto sbagliando qualcosa io nella configurazione?

**Comment by m3m3nto** [comment]
Bisognerebbe riprodurre tutto il flow, e non ho properties con trailing slash su GSC. Appena possibile faccio comunque qualche prova.

**Comment by m3m3nto** [comment]
Quindi giusto per fare chiarezza, nella configurazione hai inserito:

https://www.miosito.com/

e la proprietà su GSC è la medesima?

**Comment by C0nrad92** [comment]
Esattamente. La stessa identica stringa che ho su GSC.

Però ho notato che un altro sito di cui sono proprietario su GSC ha il trailing slash, ma ho inserito il dominio su GIAA senza e non da problemi.
Mentre se adesso metto il dominio senza trailing slash mi dice che non sono proprietario... non vorrei aver dimenticato qualche passaggio, ma non credo. Farò un altro check appena ho un pò di tempo!

Non so più che pesci prendere! :D

Grazie mille ancora per la disponibilità!

**Comment by m3m3nto** [comment]
Ciao, ho pushato un workaround quick & dirty :|
Purtroppo non ho molto tempo.
Prova a clonare direttamente il repo, al limite creo un tag se mi dici che è ok.
Grazie per la segnalazione!!

**Comment by C0nrad92** [comment]
Ciao, ho provato con il nuovo update, ma come puoi vedere dalle immagini non è cambiato molto.

Ho controllato più volte di essere proprietario della versione https su GSC, l'indirizzo email proprietario corrisponde a quello dell'account di servizio, quindi non dovrebbero esserci problemi di proprietà del dominio...
Eppure GIAA fa lo switch ad http (versione che non ho nemmeno su GSC) e quindi mi dice che non sono proprietario :/

![GIAA - img1](https://user-images.githubusercontent.com/72385123/95429493-c3ac2200-094a-11eb-8874-5f7fdd646c9f.jpg)
![GIAA - img2](https://user-images.githubusercontent.com/72385123/95429495-c4dd4f00-094a-11eb-8084-530b8c05c946.jpg)

**Comment by m3m3nto** [comment]
Lo script segue i redirect, possibile che per qualche motivo ce ne sia uno da https a http?
Il 403 che viene ritornato è il messaggio di errore della Indexing API, ma il redirect se presente viene eseguito da GIAA.

**Comment by C0nrad92** [comment]
No, tutte le pagine sono in https, e inserendo inserendo nel browser un link in http fa un redirect 307 alla pagina https...

E non è semplice come pensavo fargli seguire un redirect 307: 
![image](https://user-images.githubusercontent.com/72385123/95573853-5ffa2580-0a2c-11eb-9cd6-addf75574d14.png)

quindi boh!

**Comment by m3m3nto** [comment]
L'account su google console è lo stesso per tutti i siti che stai utilizzando con GIAA?

**Comment by C0nrad92** [comment]
Questo sito e un altro sono sullo stesso account GSC, ma l'altro sito non da problemi.

**Comment by m3m3nto** [comment]
Scusami intendevo dire se anche questo sito ha come l'altro assegnata la ownership all'account di servizio registrato sulla console delle API di google.
Comunque ho pushato la modifica per gestire il 307, anche se dovrei poter verificare la modalità di risposta del tuo server.
Riesco a fare una richiesta con Curl tipo questa?

curl -v -I https://www.miosito.com/

magari usando un tool online come: https://reqbin.com/curl

e mandarmi uno screenshot tipo questo?

![2020-10-12_13-04](https://user-images.githubusercontent.com/51357/95739931-e06a9180-0c8b-11eb-8984-0098143af559.png)

**Comment by C0nrad92** [comment]
Ah-ha! qualcosa di strano succede di sicuro!

Ho scaricato l'ultima versione, in grado di gestire il 307. 
Adesso se inserisco il link in "https://" la prima volta mi ha fatto una richiesta di update al sito in "http://" e mi ha detto che non ho i permessi e che non sono proprietario, poi riprovando ha fatto delle richieste solo allo "slug", il link senza il dominio, dandomi un errore 400 "Invalid attribute. 'url' is not in standard URL format: "/../../...;
![slug](https://user-images.githubusercontent.com/72385123/95829166-33008800-0d36-11eb-8359-f316d8add6d3.jpg)

mentre se inserisco il link in "http://" va a fare l'update del link in https! 

Quindi.. beh.. il redirect 307 funziona! 👍 

Grazie mille per adesso! :)
